app/models/fact_value.rb

```diff
@@ -9,7 +9,7 @@ class FactValue < ActiveRecord::Base
 
   has_one :parent_fact_name, :through => :fact_name, :source => :parent
 
-  scoped_search :on => :value, :in_key=> :fact_name, :on_key=> :name, :rename => :facts, :complete_value => true, :only_explicit => true
+  scoped_search :on => :value, :in_key=> :fact_name, :on_key=> :name, :rename => :facts, :complete_value => true, :only_explicit => true, :ext_method => :search_cast_facts
   scoped_search :on => :value, :default_order => true
```

Comment:
It's a little hacky, but would work I suppose.

The alternative I was thinking of was in search_cast_facts, to do:

<pre>
def self.search_cast_facts(key, operator, value)
  {
    :condition => "fact_names.name = '#{key.split('.')[1]}' AND #{cast_facts(key,operator,value)}",
    :includes  => :fact_name,
  }
end
</pre>

And return a string from cast_facts (or change to a more complex merge).